kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/bitwise-simplification.k

```diff
@@ -47,36 +48,28 @@ module BITWISE-SIMPLIFICATION [symbolic]
     rule [bitwise-or-lt-pow256]: (X |Int Y) <Int pow256 => true requires 0 <=Int X andBool 0 <=Int Y andBool X <Int pow256 andBool Y <Int pow256 [simplification]
 
-   // We should probably generalize the simplifications below for relevant powers of two.
-    rule notMaxUInt5   &Int X => 0 requires #rangeUInt(  5, X) [simplification]
-    rule notMaxUInt8   &Int X => 0 requires #rangeUInt(  8, X) [simplification]
-    rule notMaxUInt16  &Int X => 0 requires #rangeUInt( 16, X) [simplification]
-    rule notMaxUInt32  &Int X => 0 requires #rangeUInt( 32, X) [simplification]
-    rule notMaxUInt64  &Int X => 0 requires #rangeUInt( 64, X) [simplification]
-    rule notMaxUInt96  &Int X => 0 requires #rangeUInt( 96, X) [simplification]
-    rule notMaxUInt128 &Int X => 0 requires #rangeUInt(128, X) [simplification]
-    rule notMaxUInt160 &Int X => 0 requires #rangeUInt(160, X) [simplification]
-    rule notMaxUInt192 &Int X => 0 requires #rangeUInt(192, X) [simplification]
-    rule notMaxUInt208 &Int X => 0 requires #rangeUInt(208, X) [simplification]
-    rule notMaxUInt224 &Int X => 0 requires #rangeUInt(224, X) [simplification]
-    rule notMaxUInt240 &Int X => 0 requires #rangeUInt(240, X) [simplification]
-    rule notMaxUInt248 &Int X => 0 requires #rangeUInt(248, X) [simplification]
-
-    rule notMaxUInt240 &Int (X <<Int 240) => X <<Int 240 requires #rangeUInt(16, X) [simplification]
-    rule notMaxUInt248 &Int (X <<Int 248) => X <<Int 248 requires #rangeUInt( 8, X) [simplification]
-
-    rule maxUInt8   &Int (X |Int (notMaxUInt8   &Int _)) => maxUInt8   &Int X [simplification]
-    rule maxUInt16  &Int (X |Int (notMaxUInt16  &Int _)) => maxUInt16  &Int X [simplification]
-    rule maxUInt32  &Int (X |Int (notMaxUInt32  &Int _)) => maxUInt32  &Int X [simplification]
-    rule maxUInt64  &Int (X |Int (notMaxUInt64  &Int _)) => maxUInt64  &Int X [simplification]
-    rule maxUInt96  &Int (X |Int (notMaxUInt96  &Int _)) => maxUInt96  &Int X [simplification]
-    rule maxUInt128 &Int (X |Int (notMaxUInt128 &Int _)) => maxUInt128 &Int X [simplification]
-    rule maxUInt160 &Int (X |Int (notMaxUInt160 &Int _)) => maxUInt160 &Int X [simplification]
-    rule maxUInt192 &Int (X |Int (notMaxUInt192 &Int _)) => maxUInt192 &Int X [simplification]
-    rule maxUInt208 &Int (X |Int (notMaxUInt208 &Int _)) => maxUInt208 &Int X [simplification]
-    rule maxUInt224 &Int (X |Int (notMaxUInt224 &Int _)) => maxUInt224 &Int X [simplification]
-    rule maxUInt240 &Int (X |Int (notMaxUInt240 &Int _)) => maxUInt240 &Int X [simplification]
-    rule maxUInt248 &Int (X |Int (notMaxUInt248 &Int _)) => maxUInt248 &Int X [simplification]
+    // Generalization of: notMaxUIntXXX &Int Y => 0 requires #rangeUInt(XXX, Y)
+    rule [bitwise-and-notMaxUInt-zero]:
+      X &Int Y => 0
+      requires 0 <=Int X andBool X <Int pow256
```

Comment:
```suggestion
      requires #rangeUInt(256, X)
```

This should be expanded by the frontend to the above expression, so the backend will never see the `#rangeUInt(...)` symbol

Reply:
Very well, will change this and the below.